src/mock_crypto.rs

```diff
@@ -0,0 +1,323 @@
+        let new_rng = XorShiftRng::from_rng(other)?;
+        let _ = RNG.with(move |rng| rng.replace(new_rng));
+        Ok(())
+    }
+
+    /// Mock cryptographic functions.
+    pub mod crypto {
+        /// Mock signing.
+        pub mod sign {
+            use super::super::with_rng;
+            use rand::Rng;
+            use std::collections::hash_map::DefaultHasher;
+            use std::hash::Hasher;
+            use std::ops::{Index, RangeFull};
+
+            /// Number of bytes in a `PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 8;
+            /// Number of bytes in a `SecretKey`.
+            pub const SECRETKEYBYTES: usize = 8;
+            /// Number of bytes in a `Signature`.
+            pub const SIGNATUREBYTES: usize = 8;
+
+            /// Mock signing public key.
+            #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd,
+                     Serialize)]
+            pub struct PublicKey(pub [u8; PUBLICKEYBYTES]);
+
+            impl Index<RangeFull> for PublicKey {
+                type Output = [u8];
+                fn index(&self, index: RangeFull) -> &[u8] {
+                    self.0.index(index)
+                }
+            }
+
+            /// Mock signing secret key.
+            #[derive(Clone, Debug, Eq, PartialEq)]
+            pub struct SecretKey(pub [u8; SECRETKEYBYTES]);
+
+            /// Mock signature.
+            #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, Serialize, PartialEq,
+                     PartialOrd)]
+            pub struct Signature(pub [u8; SIGNATUREBYTES]);
+
+            impl AsRef<[u8]> for Signature {
+                fn as_ref(&self) -> &[u8] {
+                    &self.0
+                }
+            }
+
+            /// Generate mock public and corresponding secret key.
+            pub fn gen_keypair() -> (PublicKey, SecretKey) {
+                with_rng(|rng| {
+                    let value = rng.gen();
+                    (PublicKey(value), SecretKey(value))
+                })
+            }
+
+            /// Sign a message using the mock secret key.
+            pub fn sign_detached(m: &[u8], sk: &SecretKey) -> Signature {
+                let mut temp = m.to_vec();
+                temp.extend(&sk.0);
+                Signature(hash128(&temp))
+            }
+
+            /// Verify the mock signature against the message and the signer's mock public key.
+            pub fn verify_detached(signature: &Signature, m: &[u8], pk: &PublicKey) -> bool {
+                let mut temp = m.to_vec();
+                temp.extend(&pk.0);
+                *signature == Signature(hash128(&temp))
+            }
+
+            fn hash128(data: &[u8]) -> [u8; 8] {
+                let mut hasher = DefaultHasher::new();
+                hasher.write(data);
+
+                let hash = hasher.finish();
+                [
+                    ((hash >> 56) & 0xFF) as u8,
```

Comment:
They're not doing any harm, but I'm pretty sure we don't need any of the `& OxFF` masks here.